### ISSUE TITLE
[Request] Publish server service on ZeroConf 

### ISSUE THREAD

**Issue by xxxserxxx** [Xcreen/RestSMS#20]
Having the RestSMS service announce itself on ZeroConf would allow clients to connect to the server when the phone IP address changes -- which is frequently, in most cases.

An easy-to-use Kotlin library for ZeroConf is [available here](https://github.com/Skyost/Bonsoir).

I wrote a beta-quality desktop TUI client called [dsms](https://hg.sr.ht/~ser/dsms) that uses RestSMS and [SMS Forwarder](https://github.com/bogkonstantin/android_income_sms_gateway_webhook) together as a phone gateway to provide a chat-like SMS client. Having a way to auto-update the RestSMS endpoint when the phone's IP address changes would make the user experience much better.

**Comment by Xcreen** [comment]
I think adding this feature makes sense. 
However, I'm not sure if including the Bonsoir library is beneficial, as, to my understanding, it also requires flatter. 
In any case, adding Android's NSD should also be straightforward.

**Comment by xxxserxxx** [comment]
@Xcreen Thanks! I tried to do a little legwork, but if the NSD has built-in capability, even better. I don't do mobile development, although I _did_ try to get everything set up and implement the service announcement myself for a PR (it being only a few lines of code, and not necessarily requiring a UI change), but I never did get things set up enough that it'd compile; I got stuck for a couple hours in version incompatability loops.

Thanks for considering it!

**Comment by Xcreen** [comment]
It´s now implemented in version 2.5.0.

**Comment by xxxserxxx** [comment]
@Xcreen I pushed changes to dsms to take advantage of this. It includes trying to re-acquire a connection to RestSMS if it disappears (e.g., the mobile IP changes). It went from:

```
$ dsms -token SECRTTKN -restSmsUrl http://192.168.0.99:8090/send
```

with a restart every time the phone IP changed, to:

```
$ dsms --token SECRTTKN -a
```

and no restarts. It's a pretty huge usability improvement IMO, so thanks.
